IdentityCore/src/broker_operation/request/browser_native_message_request/MSIDBrowserNativeMessageGetTokenRequest.m

```diff
@@ -0,0 +1,137 @@
+NSString *const BROWSER_NATIVE_MESSAGE_SCOPE_KEY = @"scope";
+NSString *const BROWSER_NATIVE_MESSAGE_REDIRECT_URI_KEY = @"redirectUri";
+NSString *const BROWSER_NATIVE_MESSAGE_PROMPT_KEY = @"prompt";
+NSString *const BROWSER_NATIVE_MESSAGE_NONCE_KEY = @"nonce";
+NSString *const BROWSER_NATIVE_MESSAGE_IS_STS_KEY = @"isSts";
+NSString *const BROWSER_NATIVE_MESSAGE_STATE_KEY = @"state";
+NSString *const BROWSER_NATIVE_MESSAGE_LOGIN_HINT_KEY = @"loginHint";
+NSString *const BROWSER_NATIVE_MESSAGE_INSTANCE_AWARE_KEY = @"instance_aware";
+NSString *const BROWSER_NATIVE_MESSAGE_EXTRA_PARAMETERS_KEY = @"extraParameters";
+NSString *const BROWSER_NATIVE_MESSAGE_REQUEST_KEY = @"request";
+
+@implementation MSIDBrowserNativeMessageGetTokenRequest
+
++ (void)load
+{
+    [MSIDJsonSerializableFactory registerClass:self forClassType:self.operation];
+}
+
++ (NSString *)operation
+{
+    return @"GetToken";
+}
+
+#pragma mark - MSIDJsonSerializable
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)error
+{
+    self = [super initWithJSONDictionary:json error:error];
+    if (!self) return nil;
+    
+    if (![json msidAssertType:NSDictionary.class ofKey:BROWSER_NATIVE_MESSAGE_REQUEST_KEY required:YES error:error]) return nil;
+    NSDictionary *requestJson = json[BROWSER_NATIVE_MESSAGE_REQUEST_KEY];
+    
+    _loginHint = [requestJson msidStringObjectForKey:BROWSER_NATIVE_MESSAGE_LOGIN_HINT_KEY];
+    NSString *homeAccountId = [requestJson msidStringObjectForKey:BROWSER_NATIVE_MESSAGE_ACCOUNT_ID_KEY];
+    
+    if (homeAccountId)
+    {
+        NSArray *accountComponents = [homeAccountId componentsSeparatedByString:@"."];
+        if ([accountComponents count] != 2)
+        {
+            if (error) *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidInternalParameter, @"accountId is invalid.", nil, nil, nil, nil, nil, YES);
+            
+            return nil;
+        }
+    }
+    
+    if (homeAccountId || _loginHint)
+    {
+        _accountId = [[MSIDAccountIdentifier alloc] initWithDisplayableId:_loginHint homeAccountId:homeAccountId];
+    }
+    
+    if (![requestJson msidAssertType:NSString.class ofKey:BROWSER_NATIVE_MESSAGE_CLIENT_ID_KEY required:YES error:error]) return nil;
+    _clientId = requestJson[BROWSER_NATIVE_MESSAGE_CLIENT_ID_KEY];
+
+    NSString *authorityString = [requestJson msidStringObjectForKey:BROWSER_NATIVE_MESSAGE_AUTHORITY_KEY];
+    
+    if (authorityString)
+    {
+        NSError *localError;
+        _authority = [[MSIDAADAuthority alloc] initWithURL:[NSURL URLWithString:authorityString] rawTenant:nil context:nil error:&localError];
+        
+        if (!_authority)
+        {
+            if (localError)
+            {
+                MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"Non AAD authorities are not supported in broker - %@", MSID_PII_LOG_MASKABLE(localError));
+            }
+            
+            if (error) *error = localError;
```

Comment:
should this be inside the if localError check above ?

Reply:
it doesn't matter, if localError nil, we will assign nil to the error.

Reply:
it doesn't matter, if localError nil, we will assign nil to the error.